### DISCUSSION TITLE
Change Executioner parameters during run

### DISCUSSION THREAD

**Discussion by reverendbedford** [idaholab/moose#20548] [Q&A General] [answered]
I would like to change an executioner parameter during a simulation run (as a function of time).  Specifically I want to switch `automatic_scaling` or possibly adjust `rtol` or `atol` to deal with a radical change in the how my model is loaded, which occurs sometime pretty late in the simulation.  These parameters are not controllable, according to the docs, so I can't use the `[Control]` system.  Is there any other way to do something like this?

**Comment by hugary1995** [comment]
I recall there is a parameter to recompute the scaling factor at every step. Probably something like `compute_scaling_once = false`.

**Reply by reverendbedford** [reply]
Yes, I tried that but it's not helping. 

I'm in a situation where automatic scaling is very helpful and probably required to get convergence for a large part of the simulation time.  But then I change my BCs radically near the end of the run (contact stops occuring) and if I keep automatic scaling on (either fixed or recomputed) nonlinear convergence "bottoms out"  and won't go below the tolerances I have configured for the bulk of the simulation.

I could increase `nl_atol` or `nl_rtol`  for the whole simulation but then I would get less accuracy than I want for the bulk of the simulation.  So it would be better to be able to control those as a function of time.  I could also switch `automatic_scaling` off at some point, if that was possible.

**Comment by GiudGiud** [comment, marked as answer]
Currently not. Though there is interest for this feature, so either making a bunch of executioner parameters controllable, or making them be functions instead of single scalar/boolean values. 
MOOSE team is pretty busy rn so I dont think this will come soon unless outside contributors pitch in

**Reply by reverendbedford** [reply]
Thanks.  I take it it's not as easy as just marking those parameters as controllable :)

**Reply by GiudGiud** [reply]
Actually for some of them it very well might be. Also make sure the class holds a reference to the scalar/boolean and not a value

**Reply by hugary1995** [reply]
No, it's currently a no-go. Because `NonlinearSystemBase` doesn't have access to the outer `FEProblemSolve`. If you attempt to declare a parameter in `FEProblemSolve` as controllable, you'll have to store a reference in `FEProblemSolve`, and provide getters for others. This looks like a design issue to me.
